lib/utils/index.js

```diff
@@ -40,17 +40,16 @@ function loadLocaleMessages (pattern) {
   })
 }
 
-let localeMessages = null // locale messages
-let localeDir = null // locale dir
-
-function getLocaleMessages (localeDirectory) {
-  if (localeDir !== localeDirectory) {
-    localeDir = localeDirectory
-    localeMessages = loadLocaleMessages(localeDir)
-  } else {
-    localeMessages = localeMessages || loadLocaleMessages(localeDir)
+let localeDirCache = {} // locale messages
+
+function getLocaleMessages (context) {
+  const settings = getSettings(context)
+
+  if (settings.locale && settings.messages) {
```

Comment:
Why do you need `settings.locale` ?
We need to cover everything all locales resources.

In about configuration name of `settings.message`, I think this feature is for loading resources with complicated structure. For this reason, the name should be intuitive, such as `loadLocaleMessages`.

Reply:
Yeah this design was because I misunderstood how the library currently supports multiple locales, I've provided more details below.